### PR TITLE
feat: integrate regenerate API

### PR DESCRIPTION
**Pull request by SuZhou-Joe**
### Description
Change the implementation of regenerate from langchain to Agent framework.
Backend API reference: https://github.com/opensearch-project/ml-commons/pull/1709

**Before regenerate:**
image

**Regenrating:**
image

**After regenerate:**
image

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/routes/chat_routes.ts [hunk 1]
```diff
@@ -314,42 +314,39 @@ export function registerChatRoutes(router: IRouter, routeOptions: RoutesOptions)
...
-      const lastInputIndex = messages.findLastIndex((msg) => msg.type === 'input');
-      // Find last input message
-      const input = messages[lastInputIndex] as IInput;
-      // Take the messages before last input message as memory as regenerate will exclude the last outputs
-      messages = messages.slice(0, lastInputIndex);
-
+      /**
+       * Retrieve latest interactions from memory
+       */
+      const sessionId = sessionIdInRequestBody;
```
Comment (Hailong-am) [inline server/routes/chat_routes.ts:338]: can we rename `sessionIdInRequestBody` to `sessionId`?

File: server/services/chat/olly_chat_service.ts [hunk 2]
```diff
@@ -81,12 +67,55 @@ export class OllyChatService implements ChatService {
...
+    messages: IMessage[];
+    memoryId: string;
+  }> {
+    const { input, sessionId, rootAgentId } = payload;
+
+    const parametersPayload: {
+      question?: string;
+      verbose?: boolean;
+      memory_id?: string;
+    } = {
```
Comment (ruanyl) [inline server/services/chat/olly_chat_service.ts:89]: What about define an interface `AgentRunPayload` and then reuse it on L19, L85, L106 so we don't need to define inline types repeatedly.
```typescript
interface AgentRunPayload {
    question?: string;
    verbose?: boolean;
    memory_id?: string;
    regenerate_interaction_id?: string;
  }
```

**Review by ruanyl — APPROVED** [review]

File: server/routes/chat_routes.ts (hunk 1 above)
Comment (SuZhou-Joe) [reply server/routes/chat_routes.ts]: Sure, done for that.

File: server/services/chat/olly_chat_service.ts (hunk 2 above)
Comment (SuZhou-Joe) [reply server/services/chat/olly_chat_service.ts:89]: Agree, will do that.

**Pull request merged**